### PR TITLE
Add editorconfig validation to linting

### PR DESCRIPTION
**Pull request by rochdev**
### What does this PR do?

Add editorconfig validation to linting.

### Motivation

These rules are not currently enforced unless the extension is used and a specific file is touched.

### CODE REVIEW

**Review by tlhunter — DISMISSED** [review]

**Comment by tlhunter** [comment]
Based on all the failure I presume there's no `--fix` flag?

**Review by BridgeAR — COMMENTED** [review]
I like this. @rochdev do you mind finishing this? :)

**Comment by rochdev** [comment]
> I like this. @rochdev do you mind finishing this? :)

There was a weird edge case with LLMObs, will have to look into that before merging, but otherwise it should be mostly complete!

**Comment by BridgeAR** [comment]
This needs a rebase 😅 @rochdev

File: packages/dd-trace/test/appsec/bad-formatted-rules.json [hunk 1]
```diff
@@ -5,6 +5,5 @@
     },
     "rules": [
       {
-        
     ]
```
Comment (BridgeAR) [inline packages/dd-trace/test/appsec/bad-formatted-rules.json:9]: This has to be changed back

File: packages/dd-trace/test/llmobs/plugins/google-genai/util.spec.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        content: JSON.stringify({ outcome: 'OUTCOME_OK', output: '1\n' }),
+      }])
+    })
+
+    it('accumulates plain text streaming parts by role', () => {
+      const response = { candidates: [{ content: { parts: [{ text: 'Hello, ' }, { text: 'world!' }] } }] }
+      assert.deepStrictEqual(formatOutputMessages(response, true), [{ role: 'assistant', content: 'Hello, world!' }])
+    })
+  })
+})
```
Comment (BridgeAR) [inline packages/dd-trace/test/llmobs/plugins/google-genai/util.spec.js:65]: I am unsure in what way this landed here. I will check if that came in accidentially through a rebase

File: .editorconfig-checker.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  "AllowedContentTypes": [],
+  "PassedFiles": [],
+  "Disable": {
+    "EndOfLine": false,
+    "Indentation": false,
+    "IndentSize": false,
+    "InsertFinalNewline": false,
+    "TrimTrailingWhitespace": false,
+    "MaxLineLength": false,
+    "Charset": false
```
Comment (BridgeAR) [inline .editorconfig-checker.json]: Should some not be activated?

```suggestion
    "Indentation": false,
    "IndentSize": false,
    "MaxLineLength": false,
    "Charset": false
```

File: packages/dd-trace/test/llmobs/plugins/google-genai/util.spec.js (hunk 2 above)
Comment (BridgeAR) [reply packages/dd-trace/test/llmobs/plugins/google-genai/util.spec.js:65]: We changed a indentation of some code without a test. That is covering that line.

**Review by BridgeAR — DISMISSED** [review]
I like this!

**Review by ddyurchenko — APPROVED** [review]
